test/cloudformation/timestream_cloudformation.yaml

```diff
@@ -16,43 +18,49 @@ Resources:
     Type: AWS::Timestream::Database
     Properties:
       DatabaseName: !Sub "${DatabaseName}"
-  TimestreamTable:
-    Type: "AWS::Timestream::Table"
-    Properties:
-      DatabaseName: !Ref "TimestreamDatabase"
-      TableName: !Sub "${TableName}"
-      MagneticStoreWriteProperties:
-        EnableMagneticStoreWrites: true
-      RetentionProperties:
-        MemoryStoreRetentionPeriodInHours: "2160" # Three months of memory store retention
-        MagneticStoreRetentionPeriodInDays: "1825" # 5 years of magnetic store retention
-  TimestreamBackupPlan:
-    Type: "AWS::Backup::BackupPlan"
-    Properties:
-      BackupPlan:
-        BackupPlanName: !Sub "timestream-${DatabaseName}-${TableName}-backup"
-        BackupPlanRule:
-          - RuleName: "DailyBackups"
-            TargetBackupVault: "Default"
-            ScheduleExpression: "cron(0 0 ? * * *)"
-            StartWindowMinutes: "480"
-            CompletionWindowMinutes: "10080"
-            Lifecycle:
-              DeleteAfterDays: "35"
-            CopyActions:
-              - DestinationBackupVaultArn: !Sub "arn:${AWS::Partition}:backup:${BackupDisasterRecoveryRegion}:${AWS::AccountId}:backup-vault:Default"
+  'Fn::ForEach::Table':
+    - Name
+    - !Ref TableName
+    - 'TimestreamTable${Name}':
+        Type: "AWS::Timestream::Table"
+        Properties:
+          DatabaseName: !Ref "TimestreamDatabase"
+          TableName: !Ref "Name"
+          MagneticStoreWriteProperties:
+            EnableMagneticStoreWrites: true
+          RetentionProperties:
+            MemoryStoreRetentionPeriodInHours: "2160" # Three months of memory store retention
+            MagneticStoreRetentionPeriodInDays: "1825" # 5 years of magnetic store retention   
+      'TimestreamBackupPlan${Name}':
+        Type: "AWS::Backup::BackupPlan"
+        Properties:
```

Comment:
Is it possible for us to have a single backupPlan that references multiple resources so that we don't have to iterate through _every_ resource here?